meetings/2021-07/july-15.md

```diff
@@ -0,0 +1,734 @@
+
+ JWK:I guess that must be able to mock otherwise it might break visualization.
+
+KG: Right. So the assumption is that you are running first so that you can save off a copy of getOriginals that no one else will thereafter be in a position to modify.
+
+JWK: if they are running with why need you can just grab those by yourself
+
+JHD: I might Define a function first by the time that functions run a bunch of add code might have run or user extensions in their browser and mess with stuff and the in order for the the body of the function to be robust, I have to have cashed everything that it's going to use, and in to protect methods, I have to call bound to them in a way that they don't rely on.collared. Fine. Being present at runtime. And there's, I can already do this. I have a get intrinsic library that does this, but having it built into the platform, just like the Realms, proposal would make things a lot simpler and harder to get wrong.
+
+JWK: The realms use case seems much larger than getIntrinsics. It seems like a very limited use case. I don't know if it's worth it to do that.
+
+JHD: Yes, it's a fair point that there is a small number of people that may utilize that API, but I mean the that is a significant percentage of npm, download traffic already uses my get intrinsic Library so like it's the end users will will benefit from that. A lot of end users will benefit from the code even if they don't use it directly while. I think this may be true of realms.
+
+JWK: One of the use cases I can imagine is that I can get the async function prototype without actually using an async function literal or that might not be able to ship to some old browsers.
+
+JHD: Could you rephrase that question? I'm not sure. I understood 
+
+JWK: some intrinsics require some newer syntax reach, for example, async function. AsyncGeneratorPrototype requires an async generator literal to get. And if we ship it, it breaks on the old browser.
+
+JHD: True, and my get intrinsic Library, currently handles that by using eval to evaluate. The syntax is such that it won't fail to parse an older browsers, but it will still work into her once, like, eval inside a try catch. It's gross. I would love to not have to ship that and, you know, be able to work in CSP environment, for example, but like the yeah, I mean so so then one day it is getting the native get intrinsic function is at this point already has all of the current syntax. So like that was alright, for a get intrinsic polyfill to solve which way. I 
```

Comment:
```suggestion
JHD: True, and my `get-intrinsic` library currently handles that by using `eval` to evaluate. The syntax is such that it won't fail to parse an older browsers, but it will still work with `eval` inside a try catch. It's gross. I would love to not have to ship that and, you know, be able to work in CSP environments, for example, but like the yeah, I mean so so then one day it is getting the native get intrinsic function is at this point already has all of the current syntax. So like that was alright, for a get intrinsic polyfill to solve which way.
```
i kind of trailed off fixing the end, but i think this still improves the beginning